openai/client.py

```diff
@@ -0,0 +1,401 @@
+import sys
+import time
+
+from typing import Union, Dict, Any, cast, Iterable, BinaryIO
+if sys.version_info >= (3, 8):
+    from typing import Literal, AsyncIterable
+else:
+    from typing_extensions import Literal, AsyncIterable
```

Comment:
nit: typing extensions does the version check [for you](https://github.com/python/typing_extensions/blob/main/src/typing_extensions.py#L321) and we should import AsyncIterable from collections.abc if we can

```suggestion
from typing_extensions import Literal
if sys.version_info >= (3, 9):
    from collections.abc import AsyncIterable
else:
    from typing import AsyncIterable
```

Reply:
Yeah, the reason I did the "outside" version check is because I think openai has typing_extension as an optional install for python 3.7 only (and uses the version check pattern elsewhere)... I'll double-check because there is a distinct possibility that I am misremembering :).